base/float.jl

```diff
@@ -354,9 +354,10 @@ trunc(::Type{Unsigned}, x::IEEEFloat) = trunc(UInt,x)
 trunc(::Type{Integer}, x::IEEEFloat) = trunc(Int,x)
 
 # fallbacks
-floor(::Type{T}, x::AbstractFloat) where {T<:Integer} = trunc(T,round(x, RoundDown))
-ceil(::Type{T}, x::AbstractFloat) where {T<:Integer} = trunc(T,round(x, RoundUp))
-round(::Type{T}, x::AbstractFloat) where {T<:Integer} = trunc(T,round(x, RoundNearest))
+trunc(::Type{T}, x::Real) where {T} = round(T, x, RoundToZero)
+floor(::Type{T}, x::Real) where {T} = trunc(T,round(x, RoundDown))
+ceil(::Type{T}, x::Real) where {T} = trunc(T,round(x, RoundUp))
+round(::Type{T}, x::Real) where {T} = trunc(T,round(x, RoundNearest))
```

Comment:
These should go in floatfuncs.jl by https://github.com/JuliaLang/julia/blob/31d4c22f29e9dfa7cbe7d3c4f5212c44c5df7baf/base/floatfuncs.jl#L153-L155
and also pass on keyword arguments.
<!--- follow a similar pattern. This PRs current approach of dispatching directly to round and trunc rather than three argument `round` prevents dispatch specialized to all three arguments (e.g. `round(Int, 1//3)` no longer dispatches to https://github.com/JuliaLang/julia/blob/31d4c22f29e9dfa7cbe7d3c4f5212c44c5df7baf/base/rational.jl#L477). Perhaps we should dispatch to the three argument `round` and add a fallback definition to round and trunc.
```julia
round(::Type{T}, x::Real, rm::RoundingMode; kws...) where T = convert(T, round(x, rm; kws...))
``` --->

I also think `convert` is more appropriate than `trunc` in general. We use `trunc` instead of `convert` for `IEEEFloat`s because it is faster. Perhaps we have these functions dispatch to a three argument `round`, have three argument round fallback to `convert(T, round(x, rm))`, and define three argument round for floats to use `trunc`.

The reason we can't merge as is and make those changes later is that exporting these new generic methods is a feature and changing them would be breaking.

Reply:
Hey, sorry for dropping the ball on this, I caught up with some things and completely forgot! I think I've implemented these suggestions now. The three argument `round` for floats already uses `trunc`, I believe:
https://github.com/JuliaLang/julia/blob/5e4ecf19f1d0e2ccce7ce9e87456a771918d5a18/base/floatfuncs.jl#L122-L125
so hopefully the current iteration of the code does what was intended 😅